### PR TITLE
more doxygen work

### LINKED ISSUE #78: add documentation info on main landing page
The main page of the ip library is blank: https://noaa-emc.github.io/NCEPLIBS-ip/

We need some overview on the main page, with links to important functions...

### PR DESCRIPTION
**Pull request by edwardhartnett**
@GeorgeGayno-NOAA most I need your input on file ip_equid_cylind_grid_mod.F90.

Did I correctly identify the GRIB section as 3.0?

We need to fill in all the ??? in file ip_equid_cylind_grid_mod.F90.

Fixes #78 
Part of #97

### CODE REVIEW

File: src/ip_equid_cylind_grid_mod.F90 [hunk 1]
```diff
@@ -19,7 +24,7 @@ module ip_equid_cylind_grid_mod
   public :: ip_equid_cylind_grid
 
   type, extends(ip_grid) :: ip_equid_cylind_grid
-     real :: hi !< ???
+     real :: hi !< ???
```
Comment (GeorgeGayno-NOAA) [inline src/ip_equid_cylind_grid_mod.F90]: hi - Scan mode in the 'i' direction. GRIB2, Section 3, octet 72.
rlat1 - Latitude of first grid point. GRIB2, Section 3, octets 47-50.
rlon1 - Longitude of first grid point. GRIB2, Section 3, octets 51-54.
rlat2 - Latitude of last grid point. GRIB2, Section 3, octets 56-59.

**Comment by edwardhartnett** [comment]
@GeorgeGayno-NOAA please review...

**Review by GeorgeGayno-NOAA — CHANGES_REQUESTED** [review]
I defined some arguments in ip_equid_cyclind_grid_mod.F90. I still see ???

File: src/ip_equid_cylind_grid_mod.F90 (hunk 1 above)
Comment (edwardhartnett) [reply src/ip_equid_cylind_grid_mod.F90]: Are these the same definitions in all the grid_mod files in which they appear?

Comment (edwardhartnett) [reply src/ip_equid_cylind_grid_mod.F90]: Are dlat and dlon the same as di and dj here: https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_temp3-0.shtml

Comment (GeorgeGayno-NOAA) [reply src/ip_equid_cylind_grid_mod.F90]: Yes. For all map projections, the index of the IGDTMPL array should correspond to the order in which the   map grid information is listed in the grib2 document table.

**Review by GeorgeGayno-NOAA — APPROVED** [review]

**Pull request merged**